avogadro/core/layermanager.h

```diff
@@ -0,0 +1,121 @@
+/******************************************************************************
+  This source file is part of the Avogadro project.
+  This source code is released under the 3-Clause BSD License, (see "LICENSE").
+******************************************************************************/
+
+#ifndef AVOGADRO_CORE_LAYERMANAGER_H
+#define AVOGADRO_CORE_LAYERMANAGER_H
+
+#include "avogadrocore.h"
+
+#include "array.h"
+#include "layer.h"
+
+#include <cassert>
+#include <map>
+#include <memory>
+#include <vector>
+
+namespace Avogadro {
+namespace Core {
+
+class Molecule;
+
+/**
+ * @class LayerData layermanager.h <avogadro/core/layermanager.h>
+ * @brief Interface to store layer data structure.
+ */
+struct LayerData
+{
+  LayerData(std::string save = "") { load(save); }
+
+  /** save custom data, base save should never be called */
+  virtual std::string save()
```

Comment:
I'm not happy with this, is there anything in `Avogadro:IO` I could extend to? or any better idea?

Reply:
I mean, this is basically `serialize()` and `deserialize()` - you're not actually saving, you're just generating a string that represents the current layer configuration.

You can be sure the base method isn't called by making it a pure virtual method.

Reply:
I can't use pure virtual method. I create `LayerData` objects in `cjsonformat.cpp` and then it's serialized in `pluginlayermanager.h` where I know the class type. But the renaming gives me some comfort .